armi/reactor/cores.py

```diff
@@ -563,11 +563,15 @@ def add(self, a, spatialLocator=None):
                 "".format(aName, self.assembliesByName[aName], a, self.r.p.maxAssemNum)
             )
             raise RuntimeError("Core already contains an assembly with the same name.")
+
         self.assembliesByName[aName] = a
         for b in a:
             self.blocksByName[b.getName()] = b
 
         if self.geomType == geometry.GeomType.HEX:
+            cornerUp = self.spatialGrid.cornersUp
+            a.orientBlocks(cornersUp=cornerUp)
```

Comment:
`Block.spatialGrid` should be the opposite orientation of the grid on which its parent assembly lies
```suggestion
            a.orientBlocks(cornersUp=not cornerUp)
```